src/api/mod.rs

```diff
@@ -15,6 +15,7 @@ pub use color::*;
 
 use arrayvec::ArrayVec;
 use bitstream_io::*;
+use itertools::Itertools;
```

Comment:
If toml-rs takes less time than itertools I'd consider using this + serde. (bonus point: being able to feed rav1e with the configuration used)

Reply:
I'm not sure if there's a clean way of doing this with toml-rs, given that this currently shows some of the fields from `EncoderConfig`, then all the fields in `SpeedSettings`. Unless you have any suggestions?

Reply:
I wonder why some encoder config fields are omitted. We can discuss further on IRC probably.

Reply:
I chose to hide some of the fields because I thought some of them didn't make sense to show, or we already showed them. But thinking about it more, the only ones that we already show are width, height, and bit_depth. These are ones that we wouldn't want to be user-settable, but I think we can configure toml-rs via serde to hide those fields and set them to a default value on load (so the encoder can set them properly).

Maybe it doesn't make sense for width, height, and bit_depth to be in EncoderConfig at all? That would be a larger discussion though.

Reply:
I took a quick stab at it and ended up getting a panic: `called `Result::unwrap()` on an `Err` value: ValueAfterTable` which I think is because we have nested structs in encoder config. It's probably possible to make it work, but I'm going to go ahead and land it as-is for now.

Reply:
We can refine and improve later :)